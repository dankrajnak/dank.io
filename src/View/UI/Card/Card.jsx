// @flow
import React, { type Node } from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  position: relative
  border-radius: 10px;
  box-shadow: 0 ${props => props.shadowAmount * 8}px ${props =>
  props.shadowAmount * 10}px 0 rgba(100, 100, 100, 0.5);
  width: 300px;
  height: 500px;
  overflow: hidden;
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
`;

const TitleHoder = styled.div`
  height: 90px;
  padding: 10px;
  background-color: rgba(255, 255, 255, 0.3);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  font-family: ${props => props.theme.text.fontFamily};
`;

const BackgroundHolder = styled.div`
  flex-grow: 1;
`;

export type Props = {
  title: string,
  description: string,
  background: Node,
  shadowAmount?: ?number,
};

const Card = (props: Props) => {
  //Calculate shadow amount;
  let shadow = 1;
  if (props.shadowAmount != null) {
    shadow = Math.min(Math.max(0, props.shadowAmount), 1);
  }
  return (
    <Wrapper shadowAmount={shadow}>
      <Content>
        <TitleHoder>
          <h3>{props.title}</h3>
          <p>{props.description}</p>
        </TitleHoder>
        <BackgroundHolder>{props.background}</BackgroundHolder>
      </Content>
    </Wrapper>
  );
};

export default Card;
