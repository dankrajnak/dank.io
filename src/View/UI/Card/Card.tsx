import React from "react";
import styled from "styled-components";

export const CARD_WIDTH = 300;
export const CARD_HEIGHT = 500;

const Wrapper = styled.div`
  width: ${CARD_WIDTH}px;
  div:hover {
    cursor: pointer;
  }
`;

const CardWrapper = styled.div.attrs<{ shadowAmount: number }>(
  ({ shadowAmount }) => ({
    style: {
      boxShadow: `0 ${shadowAmount * 8}px
  ${shadowAmount * 10}px 0 rgba(100, 100, 100, 0.5)`,
    },
  })
)`
  position: relative;
  border-radius: 0;

  transition: box-shadow 0.2s ease-in-out;
  &:hover {
    box-shadow: 0 8px 10px 0 rgba(100, 100, 100, 0.5);
  }
  height: ${CARD_HEIGHT}px;
  overflow: hidden;
`;

const TitleHolder = styled.div`
  margin-top: 20px;
  color: #444;
  padding: 0px 10px;
  height: 50px;
  background: white;
  text-align: right;
`;

const Description = styled.div`
  font-size: 0.8rem;
`;
const BackgroundHolder = styled.div`
  height: 100%;
  width: 100%;
`;

export interface Props {
  title: string;
  description: React.ReactNode;
  background: React.ReactNode;
  shadowAmount?: number | null;
}

const Card = (props: Props) => {
  //Calculate shadow amount
  let shadow = 1;
  if (props.shadowAmount != null) {
    shadow = Math.min(Math.max(0, props.shadowAmount), 1);
  }
  return (
    <Wrapper title={"Go to " + props.title}>
      {/* 
      // @ts-ignore */}
      <CardWrapper shadowAmount={shadow}>
        <BackgroundHolder>{props.background}</BackgroundHolder>
      </CardWrapper>
      <TitleHolder>
        {props.title}
        <Description>{props.description}</Description>
      </TitleHolder>
    </Wrapper>
  );
};

export default Card;
