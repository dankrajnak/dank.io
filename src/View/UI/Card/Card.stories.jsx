// @flow
import React from "react";
import { storiesOf } from "@storybook/react";
import Card from "./Card.jsx";

storiesOf("Card", module).add("something", () => <Card />);
