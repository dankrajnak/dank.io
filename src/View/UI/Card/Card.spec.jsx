import React from "react";
import renderer from "react-test-renderer";

import Card from "./Card.jsx";

describe("Card", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Card siteTitle="Default Starter" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
