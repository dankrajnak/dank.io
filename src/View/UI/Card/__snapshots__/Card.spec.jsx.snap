// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="sc-bdVaJa bViqtP"
>
  <div
    className="sc-bwzfXH kcGEsV"
  >
    <div
      className="sc-htpNat kzTsvD"
    >
      <h3 />
    </div>
    <div
      className="sc-bxivhb khxFsg"
    />
  </div>
</div>
`;
